module/VuFind/src/VuFind/Auth/Manager.php

```diff
@@ -649,6 +652,10 @@ public function login($request)
         ) {
             if (!$this->csrf->isValid($request->getPost()->get('csrf'))) {
                 $this->getAuth()->resetState();
+                $this->logError(
```

Comment:
I wonder if this should be logged at a different level than "error" -- to use an HTTP analogy, this is a 400 error, while the other exceptions are 500 errors -- one is related to misconfiguration or system failure, which requires urgent attention. Bad CSRF tokens are probably the result of external user actions. It might not be desirable to flood the error logs if, for example, somebody does some sort of bot-net attack sending invalid submissions to VuFind forms. What do you think?

Reply:
Yes, it makes sense, I changed it to logWarning.